src/bin/trig.rs

```diff
@@ -257,54 +211,56 @@ impl SynthLanguage for Math {
         matches!(self, Math::RealConst(_))
     }
 
-    /// Returns true if the node is in the current domain.
-    /// Useful for rule lifting.
     fn is_allowed(&self) -> bool {
```

Comment:
what's the conceptual difference between is_allowed and is_extractable?

Reply:
I think allowed is from the allowed/forbidden description of rule lifting but extractable is for knowing if a rule is "useful". They may be the same but I haven't really thought about it.